Update 18F leadership selection guide

mgwalker opened the pull request:
Many thanks to all the people who helped put this together. It's incomplete, but it feels like a good start.

(in reverse alphabetical order, because sometimes those of with late-letter last names would like to eat first):
- Ashley Wichman (@awichman)
- Victoria Bitzer Wales (@vbitzer)
- Anne Petersen (@annepetersen)
- Hanna Kane (@hannahkane)
- Kelsey Foley (@kfoley-18F)
- Carter Baxter (@tbaxter-18f)

Also the biggest possible kudos to the Diversity Guild. ❤️

Immediate next steps:
- [x] Add a link to the leadership candidate guide in the selection guide
- [x] Update the handbook index to link to both the leadership selection and candidate guides
- [x] Get a Writing Lab once-over to make sure it's the most goodest it can be
- [ ] Have Talent look it over and gives us the thumbs up or propose changes

---

Longer-term next steps:

* Work with all of TTS and Talent to harmonize and merge with the [existing hiring page](https://handbook.tts.gsa.gov/hiring/#job-prep)

mgwalker committed:
Update leadership-selection.md

mgwalker committed:
Create leadership-candidate-guide.md

mgwalker committed:
meeting invite should include a link to the candidate guide

mgwalker committed:
I don't know how Federalist links work 🤷🏼‍♂️

mgwalker committed:
maybe this'll do it!

mgwalker committed:
maybe this will update the index

mgwalker committed:
typo

Commit:
Writing lab suggestions

Commit:
Tweak punctuation

annepetersen committed:
Update leadership-selection.md

JJediny committed:
Merge pull request #2097 from 18F/rj-edits

Suggested edits to leadership selection (microrequest)

JJediny reviewed (approved):
Out of my subject area but the merge is clean

mgwalker committed:
from AlanB

mgwalker committed:
from AlanB

Commit:
edits

changed position description to job description to eliminate confusion with GSAs position descriptions which are different that the performance profiles we use that actually describe the role.

I think this is a great idea and should be applied to all of TTS.  Can I remove references to 18F so it can be used by everyone?

Commit:
Update leadership-selection.md

changes:
- position description to job description to avoid confusion on GSA official PDs
- added a note that Talent also publicizes roles 
- added that Talent will have supervisory interview guide that will be mandatory to use for supervisory interviews - that is not complete yet

afeld reviewed (approved):
Going to go ahead and merge this; subsequent edits can be made via follow-up pull requests.
